### PR TITLE
Remove unsupported jf build step from freestyle projects

### PR DESCRIPTION
**Pull request by EyalDelarea**
- [x] This pull request is created in the [jfrog/jenkins-jfrog-plugin](https://github.com/jfrog/jenkins-jfrog-plugin) repository.

jf commands as a freestyle build step is not supported.
Remove to avoid confusion.

image

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/jfrog/JfStep.java [hunk 1]
```diff
@@ -291,7 +292,10 @@ public String getDisplayName() {
 
         @Override
         public boolean isApplicable(Class<? extends AbstractProject> jobType) {
-            return true;
+            // JfStep Extends SimpleBuildStep, which applies also to FreeStyleProjects, as well as workflow jobs.
+            // To avoid errors due to unsupported inputs in FreeStyleProjects,
+            // Return applicable only for WorkflowJobProperty.
+            return WorkflowJobProperty.class.isAssignableFrom(jobType);
```
Comment (yahavi) [inline src/main/java/io/jenkins/plugins/jfrog/JfStep.java:298]: Shouldn't be this?
```java
return WorkflowJob.class.isAssignableFrom(jobType) || WorkflowJob.class.isAssignableFrom(jobType);
```
I haven't tested the solution.
Please also verify with [Multibranch Pipelines](https://www.jenkins.io/doc/book/pipeline/multibranch/).

File: src/main/java/io/jenkins/plugins/jfrog/JfStep.java [hunk 2]
```diff
@@ -291,7 +292,10 @@ public String getDisplayName() {
 
         @Override
         public boolean isApplicable(Class<? extends AbstractProject> jobType) {
-            return true;
+            // JfStep Extends SimpleBuildStep, which applies also to FreeStyleProjects, as well as workflow jobs.
+            // To avoid errors due to unsupported inputs in FreeStyleProjects,
```
Comment (yahavi) [inline src/main/java/io/jenkins/plugins/jfrog/JfStep.java:296]: Not only FreeStyleProjects, but all non-Pipelines job types

**Review by yahavi — CHANGES_REQUESTED** [review]

**Comment by EyalDelarea** [comment]
This is causing the multi-branch pipeline to crash. 
Need to fix this.

File: src/main/java/io/jenkins/plugins/jfrog/JfStep.java (hunk 1 above)
Comment (EyalDelarea) [reply src/main/java/io/jenkins/plugins/jfrog/JfStep.java:298]: It's strange, but when i debug i'm getting an error thrown, also on the main branch when it comes to multi-branch.
On the other hand, when i installed the plugin on my local Jenkins, it doesn't crash.
Maybe it's related to the Jenkins version.
It needs to be tested.

**Comment by EyalDelarea** [comment]
was fixed due https://github.com/jfrog/jenkins-jfrog-plugin/pull/113

**Pull request closed without merging**